### ISSUE TITLE
Jinja2 If statement doesn't seem to be getting properly parsed in molecule playbook test

### ISSUE THREAD

**Issue by mattthehearn** [metacloud/molecule#1338]
# Issue Type

- Bug report

# Molecule and Ansible details

```
ansible 2.4.2.0
  config file = /home/mhearn/git/ansible-site/ansible.cfg
  configured module search path = [u'/home/mhearn/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
```

```
molecule, version 2.7.0
```

Molecule installation method (one of):

- pip

Ansible installation method (one of):6 bo

- pip

# Desired Behavior
I'm attempting to use molecule to test a playbook that uses a number of roles, each of which is pulled in using galaxy to handle the dependencies.  One role, that standardizes the UID and GID of the ansible group and user, test successfully by itself with molecule, but fails when being tested as part of the playbook.

The role itself contains some fudgery to work in vagrant; I had standardized on using 500 as the UID and GID for my ansible user many moons ago before I started using molecule, but in the vagrant centos6 box I pull, that UID is already in use for the vagrant user.  So my testing is successful on vagrant boxes, I use a Jinja2 if statement to kludge my way past the problem:

```
- name: create ansible group
  group:
    name: 'ansible'
    gid: "{{ 5000 if (ansible_virtualization_type == 'virtualbox') else 500 }}"
```

I would expect that when it runs in any virtualbox virtualized environment, it would attempt to use GID 5000 to create the group, but...

# Actual Behaviour (Bug report only)

It doesn't, it complains about GID 500 being in use:

```
<127.0.0.1> (0, '\r\n{"msg": "groupadd: GID \'500\' already exists\\n", "failed": true, "name": "ansible", "invocation": {"module_args": {"state": "present", "gid": "500", "name": "ansible", "system": false}}}\r\n', 'X11 forwarding re
quest failed\r\nShared connection to 127.0.0.1 closed.\r\n')
    fatal: [centos6-site]: FAILED! => {
        "changed": false, 
        "invocation": {
            "module_args": {
                "gid": "500", 
                "name": "ansible", 
                "state": "present", 
                "system": false
            }
        }, 
        "msg": "groupadd: GID '500' already exists\n", 
        "name": "ansible"
    }
```

I honestly don't know if the problem is Molecule, Ansible, Jinja2, etc.  But I've been beating my head against the wall to try and solve it for 2 days without any luck.  Any idea what obvious thing I'm missing?

**Comment by dhoppeIT** [comment]
Can you test it with the following code? 

```
- name: create ansible group
  group:
    name: 'ansible'
    gid: "{{ (ansible_virtualization_type == 'virtualbox') | ternary(5000, 500) }}"
```

That would rule out at least a problem with Ansible or Jinja2.

**Comment by mattthehearn** [comment]
Seems like the same deal:
```
    <127.0.0.1> (0, '\r\n{"msg": "groupadd: GID \'500\' already exists\\n", "failed": true, "name": "ansible", "invocation": {"module_args": {"state": "present", "gid": "500", "name": "ansible", "system": false}}}\r\n', 'X11 forwarding re
quest failed\r\nShared connection to 127.0.0.1 closed.\r\n')
    fatal: [centos6-site]: FAILED! => {
        "changed": false, 
        "invocation": {
            "module_args": {
                "gid": "500", 
                "name": "ansible", 
                "state": "present", 
                "system": false
            }
        }, 
        "msg": "groupadd: GID '500' already exists\n", 
        "name": "ansible"
    }
```

I also tried filtering by IP (since vagrant instances by default use "10.0.2.x" addresses that don't exist anywhere else in my environment), but got the same result.

It's weird, because it works fine when I test the role by itself; it only fails when I test a playbook that uses the role.  It's almost as if it's not actually processing any facts, or discards them before it runs that role's tasks.  I may try including some "extra-vars" settings in the molecule.yml file to identify that I'm running a molecule test, and it's okay to use an alternate gid/uid for the ansible user.

**Comment by retr0h** [comment]
This isn't a problem with Molecule.  Maybe it is a problem with your version of Ansible?

```
[jodewey:~/git/molecule_2/test/scenarios/driver/vagrant] [molecule2-env-2.7.13] master(+5/-2)+ ± ansible --version
ansible 2.5.1
  config file = None
  configured module search path = [u'/Users/jodewey/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/jodewey/.pyenv/versions/2.7.13/envs/molecule2-env-2.7.13/lib/python2.7/site-packages/ansible
  executable location = /Users/jodewey/.pyenv/versions/molecule2-env-2.7.13/bin/ansible
  python version = 2.7.13 (default, May  4 2017, 15:45:50) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

```
    TASK [debug] *******************************************************************
    ok: [instance] => {
        "ansible_virtualization_type": "virtualbox"
    }
```

**Comment by mattthehearn** [comment]
I figured it out, and as I suspected all along, I'm an idiot.  The problem is that I didn't tell galaxy where to put the roles, so it was putting it in the molecule/vagrant/.molecule directory as it should.  But then I told it to run a custom playbook; instead of molecule/vagrant/playbook.yml, it's running "../../site.yml", the playbook that I run when I build new VMs, which is defaulting to using the regular top-leve roles directory.

I changed the galaxy step to look like this:
```
 dependency:
  name: galaxy
  options:
    force: true
    role-file: roles/requirements.yml
    roles-path: roles
```

That roles-path line fixed me.  Thanks!!!
